### ISSUE TITLE
Recursively_convert_to_json_serializable doesn't check all numpy types

### ISSUE THREAD

**Issue by jpcampb2** [fivetran/great_expectations#265]
...which it says it does in the comments. The two should be harmonized.

**Comment by ccnobbli** [comment]
I'll take a look at this, found some good documentation for numpy types and type checking.

**Comment by ccnobbli** [comment]
Found some helpful stuff here:
https://docs.scipy.org/doc/numpy-1.13.0/user/basics.types.html - base type testing
   `np.issubdtype(type(test_obj), np.basetype)` evaluates to true if np data is a subtype of the basetype, e.g. `np.int32` would evaluate true if you specified basetype as `np.integer`.

and here: 
https://github.com/numpy/numpy/pull/9505 - info about FutureWarnings from numpy
   this motivated some of my decisions like to use `np.complexfloating` as a basetype instead of `np.complex`
  

I added support for `np.bool`, `np.float`, `np.integer`, `np.complex`, and their respective subtypes. Testing shows that `np.unicode` for strings also works - although I am sure there are edge cases that I am unaware of.
